client/checkout/blocks/saved-token-handler.js

```diff
@@ -1,15 +1,40 @@
 /**
  * Internal dependencies
  */
+import { useEffect } from 'react';
 import { usePaymentCompleteHandler } from './hooks';
+import { select } from '@wordpress/data';
+
+const { getPaymentMethodData } = select( 'wc/store/payment' );
```

Comment:
I'd recommend changing this to `useSelect` and actually, I just re-read the [Best practices for using useSelect() from @wordpress/data](https://jardasn.blog/2021/01/22/some-best-practices-for-using-useselect-from-wordpress-data/). It says to `Always call selector functions inside the callback`.

The new code would look like so:

```tsx
const paymentMethodData = useSelect( ( select ) => {
  const store = select('wc/store/payment' );
  return store.getPaymentMethodData();
} );
```

If you do this, then you'd probably want to make the following changes to the `useEffect`

```tsx
useEffect( () => {
        /* return here because when the useEffect reruns we want to unsubscribe the onPaymentSetup so we don't have multiple. */
		return onPaymentSetup( async () => {
			const fraudPreventionToken = document
				.querySelector( '#wcpay-fraud-prevention-token' )
				?.getAttribute( 'value' );

			return {
				type: 'success',
				meta: {
					paymentMethodData: {
						...paymentMethodData,
						'wcpay-fraud-prevention-token':
							fraudPreventionToken ?? '',
					},
				},
			};
		} );
	}, [ onPaymentSetup, paymentMethodData ] );
```

Reply:
Thanks for the review @opr, I'll do the changes you proposed.

Reply:
Addressed in [1915aa1](https://github.com/Automattic/woocommerce-payments/pull/7701/commits/1915aa1624029d53561a6bc028f00d7875517601).